server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -355,6 +365,10 @@ private static Metadata metadataForSnapshot(SnapshotsInProgress.Entry snapshot,
                     builder.put(indexMetadata, false);
                 }
             }
+
+            Map<String, DataStream> dataStreams = new HashMap<>(metadata.dataStreams());
+            dataStreams.keySet().removeIf(ds -> snapshot.dataStreams().contains(ds) == false);
```

Comment:
This look potentially broken. can we remove a data stream while it is in the process of being snapshotted? (looks like it) If so, then we're failing without error if someone concurrently removes (while the data nodes are busy doing their work) and before we get here it?

Reply:
You are right, I changed to similar check as above for indices